src/wallet/wallet.h

```diff
@@ -957,7 +968,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     bool TopUpKeyPool(unsigned int kpSize = 0);
     void ReserveKeyFromKeyPool(int64_t& nIndex, CKeyPool& keypool, bool internal);
     void KeepKey(int64_t nIndex);
-    void ReturnKey(int64_t nIndex);
+    void ReturnKey(int64_t nIndex, const CKeyID &keyid);
```

Comment:
why ? I mean KeypoolCache is (int, KeyId) anyway. It is just strange to pass them as separate parameters here.